pkg/controller/compliance/compliance_controller.go

```diff
@@ -187,6 +200,25 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	if !r.hasLicenseWatch {
+		r.wg.Add(1)
+
+		go func() {
+			defer close(r.ready)
+			for {
```

Comment:
Why do you need to run this in a goroutine if we're going to wait for it to complete in the Reconcile function anyway? Couldn't we just remove the `go func() {}` wrapper and just wait for the license watch to complete?

I was originally imagining that we wouldn't block Reconcile(), though - rather, we'd just have a check that would set degraded and return if not ready, and another goroutine which would establish the watch.

Reply:
I guess the problem is that the interface for a Reconciler doesn't expose a way to mark as ready, and so we need to do the "marking as ready" part within the Reconciler itself?

I think we could still launch this goroutine at the same time we launch the `WaitToAddLicenseKeyWatch` routine, though.It could mark one or more ready bits on the reconciler.

And then, the `Reconcile()` function just needs to do something like this:

```go
if reason, ready := r.IsReady(); !ready {
  r.SetDegraded(reason)
  return
}
```